### ISSUE TITLE
Should not lint on functions in list

### ISSUE THREAD

**Issue by radbasa** [Appsilon/box.linters#20]
## box.linters version

0.0.0.9006

## Source code to lint

```R
some_function <- function(x, y) {
  x$func(y)
}

ll <- list(
  func = function(i) {
    i + 1
  }
)

some_function(ll, 4)
```

## Lint command

```R
lintr::lint(code, linters = lintr::linters_with_defaults(defaults = box.linters::rhino_default_linters))
```

## Lint result

```
<text>:3:3: warning: [box_usage_linter] <package/module>$function does not exist.
  x$func(y)
  ^~~~~~
```

## Expected result

Should not lint on call to `x$func()`.

**Comment by radbasa** [comment]
Attempted to actively handle the `some_list$embedded_function()` pattern, but this proved to be beyond the scope of linting. This pattern conflicts with the `package$function()` or `module$function()` pattern. We know the valid `package$function()` and `module$function()` combinations, but a `some_list$embedded_function()` will be detected as a "non-existing package/module$function()`.

An active handler was to create a list of the functions embedded inside lists.

```R
some_list <- list(
  embedded_function = function() { }
)
```

Would have given us `c("embedded_function")`. However, this requires that the list definition is in the same current source file. If the list definition is in a different file, we cannot determine these embedded functions.

The solution used now is to lint `x$y()` if
- `x` is a known package/module name but `y` is not known. _To detect correct module/alias name, wrong function name_
- `y` is a known function (from the list of all `package$` and `module$` functions), but `x` is not known. _To detect correct function name, wrong module/alias name.
